### ISSUE TITLE
[Bug]: Add `error:NoMessage` only if it is there in the all the receive types in `alternative-receive-action`

### ISSUE THREAD

**Issue by lochana-chathura** [ballerina-platform/ballerina-lang#41968]
### Description

$subject.

Consider the below code,

```bal
function foobar() {
    boolean foo = true;

    worker w1 {
        if foo {
            10 -> w2;
        }
        20 -> w2;
    }

    worker w2 {
        int|error:NoMessage a = <- w1 | w1; // We should not enforce the `error:NoMessage` here.
    }

    _ = wait {a: w1, b: w2};
}
```

**Comment by lochana-chathura** [comment]
We also need to change the name from`error:NoMessageError` to `error:NoMessage` as the `error` prefix implies the type is of an error.

**Comment by lochana-chathura** [comment]
Fixed in #42050
